docs/spec.md

```diff
@@ -446,18 +435,19 @@ to that of the derivation (or derivations).
 
 
 
-## Traversable
-
-#### Dependencies
-
-  1. Functor
-  1. Foldable
+### Traversable
 
-#### Methods
+In the following signature `Applicative<U>` means that a value must
+not only match `Applicative` signature, but also fully support `Applicative` algebra.
 
-  1. `traverse :: (Traversable t, Applicative f) => Type t ~> (Type f, (a → f b), t a) → f (t b)`
+```js
+Traversable<T> {
+  traverse: (Applicative<U>, a => U<b>, T<a>) => U<T<b>>
+}
+```
```

Comment:
Yeah, I understand your point of view. I just don't think we should use module name as a type name. If I have a module called `Maybe`, the values it's working with won't be automatically of types `Maybe<number>`, `Maybe<boolean>` etc. It could use arrays for example under the hood (`[1]` for some, and `[]` for nothing), so the values that module works with would be of family of types `Array<something>`. Or it could be working with instances of `class Maybe {...}` then, values _would_ have types `Maybe<something>` after all, but that name "Maybe" will refer to a class name, not to a module name. 

This way of thinking seems more correct to what we're dealing with. Modules are not types. In principal some methods of a module can work with different types from another. Or we even may have an algebra which signature parametrized with two or more types.